spec/services/discord/sync_profile_service_spec.rb

```diff
@@ -0,0 +1,68 @@
+require 'rails_helper'
+
+describe Discord::SyncProfileService do
+  subject { described_class }
+
+  let(:discord_configuration) do
+    {
+      discord: {
+        bot_token: 'bot_token',
+        server_id: 'server_id',
+        default_role_ids: ['default_role']
+      }
+    }
+  end
+
+  let(:school) { create :school, configuration: discord_configuration }
+  let(:user) do
+    create :user,
+           school: school,
+           discord_user_id: Faker::Number.number(digits: 18)
+  end
+
+  let(:cohort_1) do
+    create :cohort,
+           school: school,
+           discord_role_ids: %w[role_1 role_2 role_shared]
+  end
+  let(:cohort_2) do
+    create :cohort, school: school, discord_role_ids: %w[role_3 role_4]
+  end
+
```

Comment:
We should add (bang operator) another cohort that has other roles that shouldn't get synced to the user.